processing/src/main/java/io/druid/query/Druids.java

```diff
@@ -962,6 +962,7 @@ public static ResultBuilder newResultBuilder()
     private Boolean merge;
     private Boolean lenientAggregatorMerge;
     private Map<String, Object> context;
+    private Boolean usingDefaultInterval;
```

Comment:
useDefaultInterval seems to be an unnecessary configuration, that allows to create inconsistency, if you pass useDefaultInterval=false, and querySegmentSpec which actually represents the default interval to SegmentMetadataQuery constructor.

I suggest the following plan:
 - Don't add usingDefaultInterval to this builder
 - Leave usingDefaultInterval parameter of SegmentMetadataQuery for compatibility, but ignore it, and document the fact that it is going to be removed.
 - In the constructor, set usingDefaultInterval=true if querySegmentSpec == null or querySegmentSpec is not null, and it has just one interval which is equal to the default interval.